utils/cache.js

```diff
@@ -59,25 +69,87 @@ const cachePool = (opt = { maximumSize: CACHE_SIZE_MB }) => {
     }
   };
 
-  return { get, set, hits, cacheStore };
+  return { get, set, evict, hits, cacheStore };
+};
+/**
+ * A MultiMap implementation where each key maps to set of unique values
+ *
+ */
+const cachedKeysStore = () => {
+  const keyStore = new Map();
```

Comment:
- let's add the type of `keyStore` as a jsdoc comment so that it helps others better maintain the code, what do you think?

Reply:
added.